docs/core_docs/docs/integrations/toolkits/outlook.mdx

```diff
@@ -0,0 +1,143 @@
+# Outlook Toolkit
+
+## Overview
+
+Using this toolkit, you can integrate Microsoft Graph API functionalities into your applications to manage email interactions.
+
+## Learn Microsoft Graph
+
+Microsoft Graph is a RESTful web API that enables you to access Microsoft Cloud service resources. With Microsoft Graph, you can integrate various Microsoft services and data, including users' emails, calendar, contacts, and more, into your applications.
+
+After you register your app and get authentication tokens for a user or service, you can make requests to the Microsoft Graph API.
+
+How to get authentication tokens for a user:   
+https://learn.microsoft.com/en-us/graph/auth-v2-user
+
+## App register and set environment variables
+Azure App Registration for Microsoft Outlook API Integration
+
+1. Register an Application:
+   - Go to Azure Portal.
+   - Navigate to Azure Active Directory -> App registrations.
+   - Register a new application, note down the Client ID.
+
+2. Generate Client Secret:
+   - In the application settings, go to Certificates & Secrets.
+   - Create a new client secret and note down the Client Secret.
+
+3. Configure Redirect URI:
+   - In the application settings, go to Authentication.
+   - Add a Redirect URI and note down the Redirect URI.
+
+4. Permissions and Admin Consent:
+   - In API permissions, grant necessary permissions for Microsoft Graph API or Outlook API.
+   - Grant admin consent for the added permissions.
+
+6. Environment Variables:
+   - Set the following environment variables:
+
+     ```env
+     OUTLOOK_CLIENT_ID=your_client_id
+     OUTLOOK_CLIENT_SECRET=your_client_secret
+     OUTLOOK_REDIRECT_URI=your_redirect_uri
+     OUTLOOK_ACCESS_TOKEN=your_access_token
+     OUTLOOK_REFRESH_TOKEN=your_refresh_token
+
+     ```
+
+   **Keep these values secure and avoid exposing them in public repositories.**
+
+## Setup
+
+You only need an `accessToken` to use the Outlook Toolkit, but they expire after an hour.
```

Comment:
So you need to click a link every hour to use this integration?

Reply:
Typically, with the integration, you don't need to manually click a link every hour. Instead, once you obtain an accessToken, there is usually a corresponding refreshToken provided. The AuthFlowREST class can utilize this refreshToken along with clientId, clientSecret, and redirectUri to refresh the expired accessToken automatically. This eliminates the need for manual intervention and keeps the process secure without the necessity for a continuous login or keeping a server open.

Reply:
No, I don't need to click the link every hour. If you have refreshToken and other credentials, the authFlowRefresh class will get a new valid token for you without the login step. Or if using authFlowToken, the same instance of the class only need to login once and then it will make use of the refreshToken.

Reply:
I don't fully understand from the docs -  is there a way to just get tht refresh token during setup and just use that?

Reply:
Sorry for the confusion, the athentication flow to get the refresh token is to send request with clientId, clientSecret and redirectUri, and then open the link from response and login, after that the server will receive a code, then you send the code with other credentials to request refreshToken and accessToken. The authFlowREST implements this flow, so for this class only clientId, clientSecret and redirectUri is needed in .env. But this class does not store the token in env. You can use other ways like postman to get the refreshToken. The default expiry time of refresh token is 14 days.
This video may help, https://www.youtube.com/watch?v=NAtiNpwnivI&t=309s